tests/countries/test_fiji.py

```diff
@@ -0,0 +1,310 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.countries.fiji import Fiji, FJ, FJI
+from tests.common import CommonCountryTests
+
+
+class TestFiji(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(2016, 2050)
+        super().setUpClass(Fiji, years=years)
+        cls.workday_holidays = Fiji(categories=WORKDAY, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(Fiji, FJ, FJI)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Fiji(years=2015, categories=(PUBLIC, WORKDAY)))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(2016, 2050)))
+        dt_obs = (
+            "2017-01-02",
+            "2022-01-03",
+            "2023-01-02",
+            "2028-01-03",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt_obs)
+        self.assertNoNonObservedHoliday(dt_obs)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+
+        dt = (
+            "2018-03-30",
+            "2019-04-19",
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_easter_saturday(self):
+        name = "Easter Saturday"
+
+        dt = (
+            "2018-03-31",
+            "2019-04-20",
+            "2020-04-11",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+
+        dt = (
+            "2018-04-02",
+            "2019-04-22",
+            "2020-04-13",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_girmit_day(self):
+        name = "Girmit Day"
+
+        dt = (
+            "2023-05-15",
+            "2024-05-13",
+            "2025-05-12",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2023, 2050))
+        self.assertNoHolidayName(name, range(2016, 2023))
+        self.assertNoNonObservedHoliday(dt)
```

Comment:
```suggestion
```
We use the "observed holidays" mechanism to define this holiday, but it's not actually "observed". Perhaps the implementation will change at some point.